### PR TITLE
Connecting all the data with MongoDB

### PR DESCRIPTION
**Pull request by Isaackomeza**
#### What does this PR do?

- Integrating blogs, messages, comments and users with mongodb and mongoose

#### Description of Task to be completed?

- All data can be stored in database
- Authenticate user with jwt 
- Protecting routes

#### How should this be manually tested?

- Use postman to test RESTful API

#### Any background context you want to provide?

- User who is not logged in can not post, delete or update data

#### What are the relevant pivotal tracker stories?

- https://trello.com/b/N84OdWfZ/my-capstone-trello-isaac

#### Screenshots (if appropriate)

- N/A

#### Questions:

- N/Q

### CODE REVIEW

File: .nyc_output/processinfo/e07edee4-3f16-48c0-8181-9aa014f19f17.json [hunk 1]
```diff
@@ -0,0 +1 @@
...\Backend-Blog-API\\src\\routes\\messagesRoutes.js","D:\\PROGRAMMING\\Andela\\BootCamp\\Backend-Blog-API\\src\\controllers\\messagesControllers.js","D:\\PROGRAMMING\\Andela\\BootCamp\\Backend-Blog-API\\src\\models\\messagesCollection.js","D:\\PROGRAMMING\\Andela\\BootCamp\\Backend-Blog-API\\src\\routes\\blogsRoutes.js","D:\\PROGRAMMING\\Andela\\BootCamp\\Backend-Blog-API\\src\\controllers\\blogsControllers.js","D:\\PROGRAMMING\\Andela\\BootCamp\\Backend-Blog-API\\src\\models\\blogsCollection.js","D:\\PROGRAMMING\\Andela\\BootCamp\\Backend-Blog-API\\src\\routes\\usersRoutes.js","D:\\PROGRAMMING\\Andela\\BootCamp\\Backend-Blog-API\\src\\controllers\\usersControllers.js","D:\\PROGRAMMING\\Andela\\BootCamp\\Backend-Blog-API\\src\\models\\usersCollection.js","D:\\PROGRAMMING\\Andela\\BootCamp\\Backend-Blog-API\\src\\routes\\commentsRoutes.js","D:\\PROGRAMMING\\Andela\\BootCamp\\Backend-Blog-API\\src\\controllers\\commentsControllers.js","D:\\PROGRAMMING\\Andela\\BootCamp\\Backend-Blog-API\\src\\models\\commentsCollection.js","D:\\PROGRAMMING\\Andela\\BootCamp\\Backend-Blog-API\\src\\routes\\auth.routes.js","D:\\PROGRAMMING\\Andela\\BootCamp\\Backend-Blog-API\\src\\middlewares\\auth.js"]}
```
Comment (Anguandia) [inline .nyc_output/processinfo/e07edee4-3f16-48c0-8181-9aa014f19f17.json]: do we need this here?

File: src/controllers/blogsControllers.js [hunk 2]
```diff
@@ -1,95 +1,104 @@
...
+        author: req.body.author,
+        title: req.body.title,
+        description: req.body.description
     });
-    if (blog) {
-        return res.status(200).json({
-            status: 200,
-            blog: 'blog successfully retrieved',
+    blog.save().then(result => {
+        console.log(result);
```
Comment (Anguandia) [inline src/controllers/blogsControllers.js]: we do not have to log to console here

**Review by Anguandia — COMMENTED** [review]
Please ignore the coverage directory and avoid logging to console, it's only good for your development environment

**Comment by Anguandia** [comment]
your build also seems to be failing, please check this out

**Review by Anguandia — COMMENTED** [review]
1. Ignore the coverage directory
2. Do not log to console except in development

**Comment by Isaackomeza** [comment]
Got it

**Pull request merged**